### PR TITLE
Bug fix: default Button disabled bg-color

### PR DESCRIPTION
**Pull request by langermank**
- Adds `background-color` for the default Button disabled state
- Extends disabled Storybook stories for better snapshot tests (will be more useful when [this PR](https://github.com/primer/react/pull/4352) merges)

### Rollout strategy

- [ ] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/Button/__tests__/__snapshots__/Button.test.tsx.snap [hunk 1]
```diff
@@ -871,6 +874,7 @@ exports[`Button respects the small size prop 1`] = `
 
 .c0:disabled [data-component=ButtonCounter] {
   color: inherit;
+  background-color: var(--button-default-bgColor-disabled,undefined);
```
Comment (jonrohan) [inline packages/react/src/Button/__tests__/__snapshots__/Button.test.tsx.snap:877]: This seems weird, but noticing the rest of the snapshot has similar `undefined` 🤷🏻

**Review by jonrohan — APPROVED** [review]

File: packages/react/src/Button/__tests__/__snapshots__/Button.test.tsx.snap (hunk 1 above)
Comment (langermank) [reply packages/react/src/Button/__tests__/__snapshots__/Button.test.tsx.snap:877]: Oh.. that's not good. Investigating

Comment (langermank) [reply packages/react/src/Button/__tests__/__snapshots__/Button.test.tsx.snap]: Actually, this looks like a normal thing that happens. I tested it in Storybook and the value is there. Its not great but we're getting rid of snaps soon anyways. Thanks for flagging!

File: packages/react/src/Button/styles.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export const getVariantStyles = (variant: VariantType = 'default', theme?: Theme
       '&:disabled': {
         color: 'primer.fg.disabled',
         borderColor: `var(--button-default-borderColor-disabled, ${theme?.colors.btn.border})`,
+        backgroundColor: `var(--button-default-bgColor-disabled, ${theme?.colors.input.disabledBg})`,
```
Comment (mperrotti) [inline packages/react/src/Button/styles.ts:21]: Are we sure `theme.colors.input.disabledBg` is the right path for the fallback? It's rendering as `undefined` in the Jest snapshot tests.

**Review by mperrotti — APPROVED** [review]

File: packages/react/src/Button/__tests__/__snapshots__/Button.test.tsx.snap (hunk 1 above)
Comment (mperrotti) [reply packages/react/src/Button/__tests__/__snapshots__/Button.test.tsx.snap]: I made a very similar comment. Ignore me if this is expected behavior.

File: packages/react/src/Button/styles.ts (hunk 2 above)
Comment (langermank) [reply packages/react/src/Button/styles.ts:21]: It looks like the border color above is also undefined, so I assumed there's something up with the test

**Pull request merged**